flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/IngressUtils.java

```diff
@@ -122,4 +154,27 @@ public static void setOwnerReference(HasMetadata owner, List<HasMetadata> resour
                         resource.getMetadata()
                                 .setOwnerReferences(Collections.singletonList(ownerReference)));
     }
+
+    public static URL getIngressUrl(String ingressTemplate, String name, String namespace) {
+        String template = addProtocol(ingressTemplate);
+        template = NAME_PTN.matcher(template).replaceAll(name);
+        template = NAMESPACE_PTN.matcher(template).replaceAll(namespace);
+        try {
+            return new URL(template);
+        } catch (MalformedURLException e) {
+            LOG.error(e.getMessage());
+            throw new ReconciliationException(
+                    String.format(
+                            "Unable to process the Ingress template(%s). Error: %s",
+                            ingressTemplate, e.getMessage()));
+        }
+    }
+
+    private static String addProtocol(String url) {
```

Comment:
It seems we do not use the protocol when creating the ingress. Why do we need this?

Reply:
I used the `URL` class as a parser basically to get the domain/path parts from the template (instead of writing our own parser) A valid URL needs the protocol though, so it's a hack basically.